app/core/src/main/java/com/fsck/k9/LocalKeyStoreManager.kt

```diff
@@ -7,18 +7,17 @@ import java.security.cert.CertificateException
 import java.security.cert.X509Certificate
 
 class LocalKeyStoreManager(
-        val localKeyStore: LocalKeyStore
+        private val localKeyStore: LocalKeyStore
 ) {
     /**
      * Add a new certificate for the incoming or outgoing server to the local key store.
      */
     @Throws(CertificateException::class)
     fun addCertificate(account: Account, direction: MailServerDirection, certificate: X509Certificate) {
-        val uri: Uri
-        if (direction === MailServerDirection.INCOMING) {
-            uri = Uri.parse(account.storeUri)
+        val uri = if (direction === MailServerDirection.INCOMING) {
```

Comment:
Side note: Even nicer with the `when` keyword. You can use `MailServerDirection.INCOMING` and `MailServerDirection.OUTGOING` without needing an else case. Builds will automatically fail when another enum value is added without this `when` being changed to add the missing case.